src/game-logic/cows/cow-selection.test.js

```diff
@@ -0,0 +1,113 @@
+import { act, screen } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+
+import { farmhandStub } from '../../test-utils/stubs/farmhandStub'
+import { generateCow, getCowDisplayName } from '../../utils'
+import { stageFocusType } from '../../enums'
+
+describe('cow selection', () => {
+  const cowId1 = 'foo'
+  const cowId2 = 'bar'
+  const cowStub1 = generateCow({ id: cowId1 })
+  const cowStub2 = generateCow({ id: cowId2 })
+  let cow1 = null
+  let cow2 = null
+
+  const getCow1 = async () => {
+    return (
+      await screen.findByAltText(getCowDisplayName(cowStub1, cowId1, false))
+    ).closest('.cow')
+  }
+
+  const getCow2 = async () => {
+    return (
+      await screen.findByAltText(getCowDisplayName(cowStub2, cowId2, false))
+    ).closest('.cow')
+  }
+
+  beforeEach(async () => {
+    jest.useFakeTimers()
+
+    cow1 = null
+    cow2 = null
+
+    await farmhandStub({
+      cowInventory: [cowStub1, cowStub2],
+      stageFocus: stageFocusType.COW_PEN,
+      purchasedCowPen: 1,
+
+      // Usually this gets initialized during the boot process, but it needs to
+      // be stubbed to prevent a spurious crash
+      cowForSale: generateCow(),
+      hasBooted: true,
+    })
+
+    cow1 = await getCow1()
+    cow2 = await getCow2()
+  })
+
+  afterEach(() => {
+    // Allow all cow movement timers to expire
+    act(() => {
+      jest.advanceTimersByTime(2000)
+    })
+  })
+
+  test('no cows are selected by default', () => {
+    expect(cow1.classList).not.toContain('is-selected')
+    expect(cow2.classList).not.toContain('is-selected')
```

Comment:
i'm looking forward to when we have some text to select! i think it will reduce this to 1 expect which makes me happy 😸